### PR TITLE
Revamp getAssetIcons

### PR DESCRIPTION
**Pull request by pivilartisant**
We are revamp the getAssetIcons fn 

Major Changes:
- we now only have one icon for each chain and one icon for each FT (we do not have every icon version anymore)
- we add two functions to generate needed icons: ```createNativeFT``` (should not have use case as we can just import the base icon token) and ```createBridgedFT``` (bread and butter of the feature). These functions allows us to create native FT icons and chain specific FT icons that have been bridged.
- The generated icons display is persistent across different use case and sizing 

Minor changes:
- ```getAssetIcons``` function now accepts isNative as param 
- ```getTokenIcons``` has been fully revamp and optimized to utilize the two new functions

### CODE REVIEW

**Comment by pivilartisant** [comment]
Screenshot from 2024-05-29 12-31-55
mapping use case with getAssetIcons function

**Review by Thykof — CHANGES_REQUESTED** [review]
super!

File: src/components/Icons/Svg/FT/tokenIcons.tsx [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
+  SepoliaBridged,
+} from './chainIcons';
 
-const createElement = (
-  Component: React.FC<{ size?: number }>,
+interface ChainIconProps extends ComponentPropsWithoutRef<'svg'> {
+  size?: number;
+}
+
+interface FtIconProps {
```
[the remark this answers is not in the document]
Comment (pivilartisant) [reply src/components/Icons/Svg/FT/tokenIcons.tsx:21]: good point

File: src/components/Icons/Svg/FT/tokenIcons.tsx [hunk 2]
```diff
@@ -45,19 +60,19 @@ const createElement = (
 export function getAssetIcons(
   symbol: string,
   originChainId?: number,
+  isNative = true,
   size?: number,
-  customClass = '',
```
[the remark this answers is not in the document]
Comment (pivilartisant) [reply src/components/Icons/Svg/FT/tokenIcons.tsx:49]: Oops yeah i forgot, I wanted to sync with to determine if having a custom class is really necessary here as i believe if would add additional complexity

[the remark this answers is not in the document]
Comment (Thykof) [reply src/components/Icons/Svg/FT/tokenIcons.tsx:49]: I also wonder if we need it...

[the remark this answers is not in the document]
Comment (pivilartisant) [reply src/components/Icons/Svg/FT/tokenIcons.tsx:49]: lets keep customClass out of it... components should adapt to icons if there is a need... not the contrary imo

File: src/components/Icons/Svg/FT/tokenIcons.tsx [hunk 3]
```diff
@@ -67,32 +82,47 @@ export function getAssetIcons(
...
+  if (originChainId && chainConfig[originChainId]) {
+    const ChainIcon = getChainIcon(
+      chainConfig[originChainId].native,
+      chainConfig[originChainId].bridged,
+    );
+
+    return Object.keys(tokenIcons).reduce((icons, token) => {
+      icons[token] = createBridgedFt(ChainIcon, tokenIcons[token], size);
+      return icons;
+    }, {} as { [key: string]: React.ReactNode });
```
[the remark this answers is not in the document]
Comment (pivilartisant) [reply src/components/Icons/Svg/FT/tokenIcons.tsx]: this syntax didn't quite work, but I pushed a new shorter return statement

**Review by Thykof — APPROVED** [review]

**Pull request merged**